core/src/callstate.js

```diff
@@ -1,37 +1,39 @@
-import daily from "./core.js";
-
 let state = {};
 let updateHandlers = [];
-let call = {};
 let waitingForSync = false;
+let call = null;
 const SYNC_MAX_WAITING_TIME = 7000;
 const UPDATE_MSG = "update-call-state";
 const REQUEST_MSG = "request-call-state";
 
 /* Public interface */
 export default {
-  onCallStateUpdate: function (key, callback) {
-    if (!updateHandlers[key]) {
-      updateHandlers[key] = [];
-    }
-    updateHandlers[key].push(callback);
-  },
+  afterCreateFrame,
+  instanceMethods: {
+    onCallStateUpdate: function (key, callback) {
+      if (!updateHandlers[key]) {
+        updateHandlers[key] = [];
+      }
+      updateHandlers[key].push(callback);
+    },
 
-  updateCallState: function (key, newState, broadcast = true) {
-    // used for an extension to request a state update,
-    // which will also get broadcast to everybody else on the call.
-    applyStateUpdate(key, newState);
-    if (broadcast) {
-      broadcastStateUpdate(key, newState);
-    }
-  },
+    updateCallState: function (key, newState, broadcast = true) {
```

Comment:
The more I look at this extension the more my gut tells me this should all be in iframe.js 🤔 It seems like all these methods are only being used in iFrame. Only in the miro extension do we use these functions too, but Miro could just use iFrame.

Reply:
There are some other upcoming extensions that will need this functionality to exist separately. Stay tuned :)